### PR TITLE
LSP Auto-import

### LINKED ISSUE #1991: LSP Auto-import suggestions
One of the most desiring feature I miss switched to helix is whole workspace suggestions with auto-importing on choose.

### PR DESCRIPTION
**Pull request by andreytkachenko**
Fixes #1991 Lsp auto import

### CODE REVIEW

File: helix-term/src/ui/completion.rs [hunk 1]
```diff
@@ -178,17 +178,12 @@ impl Completion {
...
-                    };
+                    let resolved_item = Self::resolve_completion_item(doc, item.clone());
 
-                    if let Some(additional_edits) = item
-                        .additional_text_edits
+                    if let Some(additional_edits) = resolved_item
                         .as_ref()
-                        .or(resolved_additional_text_edits.as_ref())
+                        .and_then(|item| item.additional_text_edits.as_ref())
+                        .or(item.additional_text_edits.as_ref())
```
Comment (pickfire) [inline helix-term/src/ui/completion.rs:186]: This seemed weird to me.

**Review by pickfire — APPROVED** [review]
Not tested but I think it looks good to me. Thanks.

**Comment by andreytkachenko** [comment]
> Looks good to me. Thanks. I just roughly tested with an import action but not sure how to invoke the auto import.

Just type `Dur` inside any function and completion list should appears. Choose `Duration (use std::time::Duration) struct` for instance and you'll notice that:
 - you typing completed to `Duration`
 - `use std::time::Duration;` added to imports

**Comment by thomkeh** [comment]
Isn't there already auto-import in helix? It definitely seems to work for me with the [pyright](https://github.com/microsoft/pyright) LS.

**Comment by andreytkachenko** [comment]
> Isn't there already auto-import in helix? It definitely seems to work for me with the [pyright](https://github.com/microsoft/pyright) LS. But maybe the rust LS doesn't support it?

There is implementation for `additionalTextEdits`. I guess `pyright` sends `additionalTextEdits` as response to `complete` request. But other LSP (such as rust-analyzer) fills `additionalTextEdits` only after `resolve`. My enables compatibilities and changes precedence order to resolved item then completion item.

File: helix-term/src/ui/completion.rs [hunk 2]
```diff
@@ -178,17 +178,12 @@ impl Completion {
                     });
 
                     // apply additional edits, mostly used to auto import unqualified types
-                    let resolved_additional_text_edits = if item.additional_text_edits.is_some() {
-                        None
-                    } else {
-                        Self::resolve_completion_item(doc, item.clone())
-                            .and_then(|item| item.additional_text_edits)
-                    };
```
Comment (archseer) [inline helix-term/src/ui/completion.rs:186]: Was this change necessary? We avoided additional lookups if the item was already fully resolved.

Comment (andreytkachenko) [reply helix-term/src/ui/completion.rs]: I agree with you, but still we need to check whether list is empty or not.

**Review by archseer — APPROVED** [review]
